FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenOperationsTest.m

```diff
@@ -106,11 +108,6 @@ - (void)setUp {
   // `FIRMessagingTokenOperation` uses `FIRInstallations` under the hood to get FIS auth token.
   // Stub `FIRInstallations` to avoid using a real object.
   [self stubInstallations];
-
-  NSString *const kHeartbeatStorageFile = @"HEARTBEAT_INFO_STORAGE";
-  GULHeartbeatDateStorage *dataStorage =
```

Comment:
I wonder if we can remove this GULHeartbeatDateStorage completely. Is it ever used? I feel like token operation just use a  @"X-firebase-client-log-type" heart beat string that I'm not sure how it is related to GULHeartbeatDataStorage

Reply:
I think the test was ported from IID. IID accesses `GULHeartbeatDateStorage` via `FIRHeartbeatInfo` to get value for "X-firebase-client-log-type". It looks still relevant to me unless I'm missing something. Yeah, I agree, it's not the cleanest test and ideally we should mock `FIRHeartbeatInfo` here instead of using real objects and modifying underlying storage. I would suggest to make this cleanup as a follow up if the PR is blocking the release, otherwise we probably can do the cleanup here. WDYT?

Reply:
To add: The `testThatTokenOperationsAuthHeaderStringMatchesCheckin` [expects `FIRHeartbeatInfo` to update the heartbeat](https://github.com/firebase/firebase-ios-sdk/pull/7995/files#diff-d970d32555eab994011dca7012f02f05445f82e8bb1eaf5f49359b06f43b20d4L393). So since we want to trigger a heartbeat update, we need to make sure `FIRHeartbeatInfo` decides to update the storage. If we don't manually remove the storage's artifacts after each test run, then `FIRHeartbeatInfo` [won't properly update the heartbeat because it will read the existing info](https://github.com/firebase/firebase-ios-sdk/blob/fe06273b691d4733be5a862f1f49d2725a8ce120/FirebaseCore/Sources/FIRHeartbeatInfo.m#L46) from a previous run and decide not to call `setHeartbeatForTag:`... which will [cause the assertion in the test to fail](https://github.com/firebase/firebase-ios-sdk/pull/7995/files#diff-d970d32555eab994011dca7012f02f05445f82e8bb1eaf5f49359b06f43b20d4L393) .

Reply:
I believe @"X-firebase-client-log-type" is [a hardcode string](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseMessaging/Sources/Token/FIRMessagingTokenFetchOperation.m#L36) and `GULHeartbeatDateStorage` is not used in the Messaging or instanceID library itself.

I'm guessing FIRURLSessionOCMockStub is the reason that is using GULHeartbeatDataStorage?

Anyway I don't want to block the release, i'm okay submitting it to get it green, but would like to understand more on this.